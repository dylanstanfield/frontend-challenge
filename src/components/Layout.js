import { styled, Typography, lighten } from '@mui/material';

// Design Credit https://dribbble.com/shots/17564792-Log-in-page-Untitled-UI?utm_source=Clipboard_Shot&utm_campaign=jordanhughes&utm_content=Log%20in%20page%20%E2%80%94%20Untitled%20UI&utm_medium=Social_Share&utm_source=Clipboard_Shot&utm_campaign=jordanhughes&utm_content=Log%20in%20page%20%E2%80%94%20Untitled%20UI&utm_medium=Social_Share

const Wrapper = styled('div')(({ theme }) => `
  width: 100vw;
  display: flex;
`);

const ContentWrapper = styled('div')(({ theme }) => `
  min-height: 100vh;
  width: 50vw;
  display: flex;
  flex-direction: column;
  gap: ${theme.spacing(8)};
  padding: ${theme.spacing(4)};

  @media (max-width: 800px) {
    width: 100vw;
  }
`);

const Title = styled(Typography)(({ theme }) => `
  font-weight: 800;
  font-size: 1.2rem;
`);

const ChildrenWrapper = styled('div')(({ theme }) => `
  flex: 1;
  width: 100%;
  max-width: 400px;
  align-self: center;
  display: flex;
  align-items: center;
`);

const HeroImageWrapper = styled('div')(({ theme }) => `
  min-height: 100vh;
  width: 50vw;
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: ${lighten(theme.palette.primary.main, 0.95)};

  @media (max-width: 800px) {
    display: none;
  }
`);

const HeroImage = styled('div')(({ theme }) => `
  background-color: ${theme.palette.primary.main};
  height: 150px;
  width: 150px;
  border-radius: 50%;
  position: relative;

  &::after {
    content: ' ';
    position: absolute;
    top: 75px;
    left: -50px;
    right: -50px;
    bottom: -50px;
    backdrop-filter: blur(15px);
  }
`);

export const Layout = ({ children }) => (
  <Wrapper>
    <ContentWrapper>
      <Title>&#9679; Upgrade Challenge</Title>
      <ChildrenWrapper>
        { children }
      </ChildrenWrapper>
      <Typography variant="caption">&#169; Upgrade Challenge 2023</Typography>
    </ContentWrapper>
    <HeroImageWrapper>
      <HeroImage />
    </HeroImageWrapper>
  </Wrapper>
)