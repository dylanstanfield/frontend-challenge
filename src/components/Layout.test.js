import { render } from "@testing-library/react";

import { Layout } from "./Layout";

describe('Layout', () => {
  it('should match snapshot', () => {
    const { container } = render(<Layout>Hello World</Layout>);
    expect(container).toMatchSnapshot();
  });
});