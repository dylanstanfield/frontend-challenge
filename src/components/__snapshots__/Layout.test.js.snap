// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should match snapshot 1`] = `
<div>
  <div
    class="css-14c4e9i"
  >
    <div
      class="css-nhrhr0"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-ixu74u-MuiTypography-root"
      >
        ● Upgrade Challenge
      </p>
      <div
        class="css-150avzy"
      >
        Hello World
      </div>
      <span
        class="MuiTypography-root MuiTypography-caption css-1sn4lm3-MuiTypography-root"
      >
        © Upgrade Challenge 2023
      </span>
    </div>
    <div
      class="css-zkjdrj"
    >
      <div
        class="css-1o8t0e3"
      />
    </div>
  </div>
</div>
`;
